sync/clients/__init__.py

```diff
@@ -15,3 +17,54 @@ def encode_json(obj: dict) -> tuple[dict, str]:
         "Content-Length": str(len(json)),
         "Content-Type": "application/json",
     }, json
+
+
+class RetryableHTTPClient:
```

Comment:
Instead of copy/pasting the retry logic to the clients we have, I figured we could just have a little base class that our clients could inherit from as appropriate.

Alternatively, these could just be functions that accept a client and a request as input.